subsys/net/lib/download_client/src/download_client.c

```diff
@@ -790,15 +789,15 @@ void download_thread(void *client, void *a, void *b)
 				rc = request_send(dl);
 				send_request = false;
 				if (rc) {
-					rc = error_evt_send(dl, ECONNRESET);
+					rc = error_evt_send(dl, -rc);
 					if (rc) {
 						/* Restart and suspend */
 						break;
 					}
 
 					rc = reconnect(dl);
 					if (rc) {
-						error_evt_send(dl, EHOSTDOWN);
+						error_evt_send(dl, -rc);
```

Comment:
The carrier library depends on specific error values from socket calls because it has different recover and retry handling depending on reason of failure. It doesn't matter which call failed. Remember this is production code, not a sample.
- If this is a `ENETDOWN` we need to bring the PDN up and retry.
- If this is a `ECONNREFUSED` we need to select a different root certificate.
- If this is a `E2BIG` then the application is misconfigured and we cannot continue.
- Etc.

Reply:
The documentation for `DOWNLOAD_CLIENT_EVT_ERROR` states "Error reason may be one of the following", so this list is not complete.

Reply:
I reverted to use `EHOSTDOWN` in the cases where a `reconnect()` is triggered because application did not handle the error from a previous error event (returned -1).

Reply:
> Remember this is production code, not a sample.

Yes, this is a shared library which is used by other applications and customers, not just the carrier lib.
I think this change should have an entry in the changelog.

Reply:
ah, apologies, i did not notice that these were just reverts.

Reply:
Actually my last comment is not entirely true, when looking at this again I have also changed send/recv error codes. This may not be needed for now, I'll have a second look at what's strictly needed to be fixed for this release.

Reply:
Because of the uncertainty changing error values for other users of the download_client I have reverted that commit from this PR. The "Report unique error codes" will be a separate PR.